main/src/main/java/cgeo/geocaching/utils/BackupUtils.java

```diff
@@ -300,7 +305,19 @@ public void restoreInternal(final Activity activityContext, final Folder backupD
         };
 
         if (database) {
-            restoreDatabaseInternal(backupDir, consumer);
+            final Folder backupTracksDir = Folder.fromFolder(backupDir, TRACKS_SUBFOLDER);
+            final List<ContentStorage.FileInformation> tracks = ContentStorage.get().list(backupTracksDir);
+            AndroidRxUtils.andThenOnUi(Schedulers.io(), () -> {
+                    final AtomicInteger tfCopied = new AtomicInteger(0);
+                    for (ContentStorage.FileInformation fi : tracks) {
+                        final Uri uri = ContentStorage.get().copy(fi.uri, Folder.fromFile(LocalStorage.getTrackfilesDir()), FileNameCreator.forName(fi.name), false);
```

Comment:
In class FolderUtils you will find methods to copy multiple files/folders at once

Reply:
Thanks for the pointer, I'll look into it.

Reply:
updated PR accordingly in both backup and restore methods